clustered/common/src/main/java/org/ehcache/clustered/common/messages/EhcacheCodec.java

```diff
@@ -34,30 +34,28 @@
 
   @Override
   public byte[] encodeMessage(EhcacheEntityMessage message) {
-    if (message.getType() == EhcacheEntityMessage.Type.LIFECYCLE_OP) {
-      return LifeCycleOpCodec.encode((LifecycleMessage) message);
-    } else {
-      return ServerStoreOpCodec.encode((ServerStoreOpMessage) message);
-    }
+    return message.encode();
```

Comment:
I agree with Louis: keeping the codecs separated from the messages makes sense in this context as it will ease backward compatibility when we start supporting multiple versions of the messages.